dove/tests/test_cmd_dove_init.rs

```diff
@@ -0,0 +1,159 @@
+            &project_name,
+            Some("dfinance"),
+            Some(address),
+            None,
+        )
+        .unwrap();
+        set_dependencies_local_move_stdlib(&project_folder);
+        project_build(&project_folder).unwrap();
+        project_remove(&project_folder);
+    }
+}
+/// $ dove init -d diem -a ###
+#[test]
+fn test_cmd_dove_init_diem_with_address() {
+    // Project name and path
+    let project_name = "demoproject_42";
+    for address in &["0x1"] {
+        let project_folder = project_start_for_init(project_name);
+        execute_dove_at(
+            &["dove", "init", "-d", "diem", "-a", address],
+            &project_folder,
+        )
+        .unwrap();
+        check_dove_toml(
+            &project_folder,
+            &project_name,
+            Some("diem"),
+            Some(address),
+            None,
+        )
+        .unwrap();
+        set_dependencies_local_move_stdlib(&project_folder);
+        project_build(&project_folder).unwrap();
+        project_remove(&project_folder);
+    }
+}
+/// $ dove init -d pont -a ###
+#[test]
+fn test_cmd_dove_init_pont_with_address() {
+    // Project name and path
+    let project_name = "demoproject_40";
+    for address in &["5CdCiQzNRZXWx7wNVCVjPMzGBFpkYHe3WKrGzd6TG97vKbnv", "0x1"] {
+        let project_folder = project_start_for_init(project_name);
+        // $ dove init -d pont -a ###
+        execute_dove_at(
+            &["dove", "init", "-d", "pont", "-a", address],
+            &project_folder,
+        )
+        .unwrap();
+        check_dove_toml(
+            &project_folder,
+            &project_name,
+            Some("pont"),
+            Some(address),
+            None,
+        )
+        .unwrap();
+        set_dependencies_local_move_stdlib(&project_folder);
+        project_build(&project_folder).unwrap();
+        project_remove(&project_folder);
+    }
+}
+/// $ dove init -d pont
+#[test]
+fn test_cmd_dove_init_pont_with_repo() {
+    // Project name and path
+    let project_name = "demoproject_45";
+    for api in &[
+        "http://demo.ru/api",
+        "https://demo.ru/api",
+        "http://127.0.0.1/api",
+        "http://localhost/api",
+        "http://localhost:8080/api",
+    ] {
+        let project_folder = project_start_for_init(project_name);
+        execute_dove_at(&["dove", "init", "-r", api], &project_folder).unwrap();
+        check_dove_toml(&project_folder, &project_name, None, None, Some(api)).unwrap();
+        set_dependencies_local_move_stdlib(&project_folder);
+        project_build(&project_folder).unwrap();
+        project_remove(&project_folder);
+    }
+}
+/// $ dove init -d incorrectdialect
+#[test]
+fn itest_cmd_dove_init_incorrect_dialect() {
+    // Project name and path
+    let project_name = "demoproject_46";
+    let project_folder = project_start_for_init(project_name);
+    assert!(
+        execute_dove_at(&["dove", "init", "-d", "incorrectdialect"], &project_folder).is_err()
```

Comment:
In this test and in the following, we should probably check if no new files or directories were created.